docs/roadmap_design.md

```diff
@@ -1,6 +1,7 @@
 # IREE Design Roadmap
+<a id="markdown-IREE%20Design%20Roadmap" name="IREE%20Design%20Roadmap"></a>
 
-<a id="markdown-iree-design-roadmap" name="iree-design-roadmap"></a>
+<!-- WARNING: DO NOT EDIT THIS FILE IN AN EDITOR WITH AUTO FORMATTING -->
```

Comment:
nit: could you mention how the TOC was generated? Was it an editor extension?

Reply:
will in vscode docs

Reply:
the extension will eat any <!-- toc --> so I can't put the link to it inline - here for now: 
https://marketplace.visualstudio.com/items?itemName=huntertran.auto-markdown-toc